arangod/Graph/GraphManager.cpp

```diff
@@ -97,18 +97,42 @@ OperationResult GraphManager::createCollection(std::string const& name, TRI_col_
                                                bool waitForSync, VPackSlice options) {
   TRI_ASSERT(colType == TRI_COL_TYPE_DOCUMENT || colType == TRI_COL_TYPE_EDGE);
 
+  auto& vocbase = ctx()->vocbase();
+  
+  VPackBuilder helper;
+  helper.openObject();
+
   if (ServerState::instance()->isCoordinator()) {
-    Result res = ShardingInfo::validateShardsAndReplicationFactor(options, ctx()->vocbase().server());
+    Result res = ShardingInfo::validateShardsAndReplicationFactor(options, vocbase.server());
     if (res.fail()) {
       return OperationResult(res);
     }
+
+    uint64_t numberOfShards = 
+      arangodb::basics::VelocyPackHelper::readNumericValue<uint64_t>(options, StaticStrings::NumberOfShards, 0);
+ 
+    if (vocbase.server().getFeature<ClusterFeature>().forceOneShard()) {
+      // force a single shard with shards distributed like "_graph"
+      helper.add(StaticStrings::NumberOfShards, VPackValue(1));
+      helper.add(StaticStrings::DistributeShardsLike, VPackValue(vocbase.shardingPrototypeName()));
+    } else if (vocbase.sharding() == "single" && 
+               options.get(StaticStrings::DistributeShardsLike).isNone() &&
+               numberOfShards <= 1) {
+      helper.add(StaticStrings::NumberOfShards, VPackValue(1));
+      helper.add(StaticStrings::DistributeShardsLike, VPackValue(vocbase.shardingPrototypeName()));
+    }
```

Comment:
Both branches of this if seem to be the same. Should we just combine the clauses?